packages/bot/src/stores/summary-store.js

```diff
@@ -0,0 +1,98 @@
+import {
+    observable,
+    action,
+    reaction }       from 'mobx';
+import { observer }  from '../utils/observer';
+
+export default class SummaryStore {
+    @observable currency        = '';
+    @observable lost_contracts  = 0;
+    @observable number_of_runs  = 0;
+    @observable total_profit    = 0;
+    @observable total_payout    = 0;
+    @observable total_stake     = 0;
+    @observable won_contracts   = 0;
+    
+    constructor(root_store) {
+        this.root_store        = root_store;
+        const { client }       = this.root_store.core;
+        this.currency          = client.currency;
+
+        observer.register('contract.status', this.onContractStatusEvent);
+    }
+
+    @action.bound
+    onContractStatusEvent(contract_status) {
+        switch (contract_status.id) {
+            case ('contract.purchase_recieved'): {
+                const { buy } = contract_status;
+                this.total_stake += buy.buy_price;
+                break;
+            }
+            case ('contract.sold'): {
+                const { contract }   = contract_status;
+                this.total_profit   += contract.profit;
+                this.number_of_runs += 1;
+
+                switch (contract.status) {
```

Comment:
Can we simplify this by 
```
if(contract.status === 'won' || (contract.status === 'sold' && contract.profit > 0)) {
 this.won_contract += 1;
 this.total_payout += contract.profit;
} else {
 this.lost_contracts += 1;
}
```

Reply:
It would reduce the number of lines, but I don't agree it would be simpler. Also there's a difference between a `won` contract and a `sold` contract (that makes a profit) is the value we add to `total_payout`.

I'll leave it as is cause it's clear to see what happens in which situation.